sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts

```diff
@@ -0,0 +1,659 @@
+    const disbursementSchedule = await db.disbursementSchedule.findOne({
+      select: {
+        id: true,
+        msfaaNumber: { id: true, msfaaNumber: true },
+      },
+      relations: {
+        msfaaNumber: true,
+        studentAssessment: true,
+      },
+      where: {
+        studentAssessment: { id: application.currentAssessment.id },
+      },
+    });
+    // Assert MSFAA Number.
+    expect(disbursementSchedule.msfaaNumber.msfaaNumber).toBe(
+      savedSFASIndividual.partTimeMSFAANumber,
+    );
+  });
+
+  it("Should create a new MSFAA number for part-time applications by creating and activating in SIMS when MSFAA is not found or invalid SFAS application offering end date.", async () => {
+    // Arrange
+    const firstLegacyApplicationStartDate = addDays(-MAX_MSFAA_VALID_DAYS + 10);
+    const firstLegacyApplicationEndDate = addDays(-MAX_MSFAA_VALID_DAYS + 2);
+
+    // Create student and save fake application with disbursements.
+    const student = await saveFakeStudent(db.dataSource);
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+
+    const savedSFASIndividual = await saveFakeSFASIndividual(db.dataSource, {
+      initialValues: { id: student.id },
+    });
+
+    await db.sfasPartTimeApplications.save(
+      createFakeSFASPartTimeApplication(
+        { individual: savedSFASIndividual },
+        {
+          initialValues: {
+            startDate: getISODateOnlyString(firstLegacyApplicationStartDate),
+            endDate: getISODateOnlyString(firstLegacyApplicationEndDate),
+          },
+        },
+      ),
+    );
+
+    // Act
+    const associateMSFAAPayload = createFakeAssociateMSFAAPayload({
+      assessmentId: application.currentAssessment.id,
+    });
+    const saveResult = await disbursementController.associateMSFAA(
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the student in SIMS.
+    const createdMSFAANumber = await db.msfaaNumber.findOne({
+      select: {
+        id: true,
+      },
+      where: {
+        student: { id: student.id },
+      },
+    });
+    // Fetch MSFAA Number for the disbursement schedule assigned to the student in SIMS.
+    const disbursementSchedule = await db.disbursementSchedule.findOne({
+      select: {
+        id: true,
+        msfaaNumber: { id: true },
+      },
+      relations: {
+        msfaaNumber: true,
+        studentAssessment: true,
+      },
+      where: {
+        studentAssessment: { id: application.currentAssessment.id },
+      },
+    });
+    // Assert MSFAA Number.
+    expect(disbursementSchedule.msfaaNumber.id).toBe(createdMSFAANumber.id);
```

Comment:
Similar to the below one, even though it is not the core of the ticket it would be good to have it due to the refcator.
https://github.com/bcgov/SIMS/pull/3817/files#r1815802662